test/test_e2e.rb

```diff
@@ -104,4 +109,27 @@ def test_spec_file_splitting
       "./spec/fast_spec.rb",
     ], queue)
   end
+
+  def test_suite_with_failures_and_fail_fast
+    queue = exec_build("failing_suite", "--fail-fast 1")
+
+    assert_equal 1, queue.fail_fast
+    assert_operator queue.unprocessed_jobs.length, :<=, 3
```

Comment:
I think this is incorrect. `unprocessed_jobs.length` should be either `1` or `2`, right? `0` and `3` are not valid values and this assertion doesn't take this fact into account.